lib/puma/client.rb

```diff
@@ -165,7 +168,7 @@ def decode_chunk(chunk)
           chunk = chunk[@partial_part_left..-1]
           @partial_part_left = 0
         else
-          @body << chunk
+          @body << chunk if @partial_part_left > 2 # don't include the last \r\n
```

Comment:
this feels brittle, because we're not actually certain that the last 2 characters are \r\n.

Reply:
I think so too but the current implementation doesn't check content: https://github.com/puma/puma/blob/675344e8609509b0d767ae7680436b3b382d8394/lib/puma/client.rb#L163

So I followed this style.

I think that robust content check should be done in a separated pull request.

Reply:
Agreed.